### ISSUE TITLE
Allow "found raw type" to upgrade to an error - Do not allow raw types

### ISSUE THREAD

**Issue by eraneverlaw** [google/error-prone#1799]
### Description of the problem / feature request:

The Java compiler issues "found raw type" warnings when a generic type is used without providing actual generic type arguments. (i.e. opting out of generics)
It would be nice if Error Prone had its own warning for this, that more importantly, can be upgraded to an error, and so prevent new code from using raw types. 

### Feature requests: what underlying problem are you trying to solve with this feature?

It's been about 16 years since Java introduced generics. It was always recommended to avoid using raw types in new code. The only reason it allowed raw types was for backwards comparability with pre-generics written code. 
Yet programmers still write NEW code that uses raw types to this day, either by mistake, or due to laziness, or ignorance of the importance of always using actual generic arguments. It would be nice to enforce this, mostly so the code is cleaner, less confusing, does not generate warnings (that are ignored,) and of course, more type-safe. 

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

`List stuff;`

### What version of Error Prone are you using?

2.3.4

###  Have you found anything relevant by searching the web?

No.

**Comment by nick-someone** [comment]
You should be able to have javac do this by adding `-Werror:+rawtypes` to the javacopts.

**Comment by tbroyer** [comment]
Wait, since when was `-Werror` upgrade from a boolean option to a complex `-Xlint`-like one?

From https://github.com/openjdk/jdk/search?q=Werror+path%3Asrc&unscoped_q=Werror+path%3Asrc, it looks like you might have conflated `Werror` with `Xlint` :thinking: 
That'd have been great though :man_shrugging:

**Comment by nick-someone** [comment]
Add, apologies, that's something that Bazel's javac main has: https://github.com/bazelbuild/bazel/blob/47f4cb6df0cb8c88da08ef7aba4c863497e5fd97/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/javac/BlazeJavacMain.java#L171

(which I guess is also helpful if eraneverlaw@ is using Bazel!)

But yeah, apologies for not noting that assumption earlier.

**Comment by cushon** [comment]
I agree being able to upgrade `rawtypes` diagnostics to errors is nice, but I'd prefer to let javac be the source of truth about rawtypes diagnostics instead of re-implementing them in Error Prone. I think that adding something like `-Werror:+rawtypes` built in to javac would be a better direction to go here.
